### ISSUE TITLE
[exportverilog] "automatic logic" variables should try to fold their initializers

### ISSUE THREAD

**Issue by lattner** [llvm/circt#1567]
We generate "automatic logic" variables in procedural blocks when we need to split expressions for various reasons, but don't fold the initializer expression into them.  Here's on example that shows the problem:

```
hw.module @foo(%reset: i1) {
   
    %true = hw.constant true

    %119 = sv.reg  : !hw.inout<i3>
    %120 = sv.reg  : !hw.inout<i3>
    %121 = sv.reg  : !hw.inout<i3>

    sv.initial {
      %504 = comb.xor %reset, %true : i1
      sv.if %504  {
        %RANDOM = sv.verbatim.expr.se "`RANDOM" : () -> i32
        %505 = comb.extract %RANDOM from 0 : (i32) -> i3
        sv.bpassign %119, %505 : i3
        %RANDOM_0 = sv.verbatim.expr.se "`RANDOM" : () -> i32
        %506 = comb.extract %RANDOM_0 from 0 : (i32) -> i3
        sv.bpassign %120, %506 : i3
        %RANDOM_1 = sv.verbatim.expr.se "`RANDOM" : () -> i32
        %507 = comb.extract %RANDOM_1 from 0 : (i32) -> i3
        sv.bpassign %121, %507 : i3
      }
    }
    
    hw.output 
  }
```

Generates:

```
module foo(
  input reset);

  reg [2:0] _T;	// a.mlir:8:12
  reg [2:0] _T_0;	// a.mlir:9:12
  reg [2:0] _T_1;	// a.mlir:10:12

  initial begin	// a.mlir:12:5
    if (~reset) begin	// a.mlir:13:14, :14:7
      automatic logic [31:0] _T_2;	// a.mlir:15:19
      automatic logic [31:0] _T_3;	// a.mlir:18:21
      automatic logic [31:0] _T_4;	// a.mlir:21:21

      _T_2 = `RANDOM;	// a.mlir:15:19
      _T = _T_2[2:0];	// a.mlir:16:16, :17:9
      _T_3 = `RANDOM;	// a.mlir:18:21
      _T_0 = _T_3[2:0];	// a.mlir:19:16, :20:9
      _T_4 = `RANDOM;	// a.mlir:21:21
      _T_1 = _T_4[2:0];	// a.mlir:22:16, :23:9
    end
  end // initial
endmodule
```

We should generate:

```
module foo(
  input reset);

  reg [2:0] _T;	// a.mlir:8:12
  reg [2:0] _T_0;	// a.mlir:9:12
  reg [2:0] _T_1;	// a.mlir:10:12

  initial begin	// a.mlir:12:5
    if (~reset) begin	// a.mlir:13:14, :14:7
      automatic logic [31:0] _T_2 = `RANDOM;
      automatic logic [31:0] _T_3 = `RANDOM;
      automatic logic [31:0] _T_4 = `RANDOM;

      _T = _T_2[2:0];	// a.mlir:16:16, :17:9
      _T_0 = _T_3[2:0];	// a.mlir:19:16, :20:9
      _T_1 = _T_4[2:0];	// a.mlir:22:16, :23:9
    end
  end // initial
endmodule
```

**Comment by lattner** [comment]
Relatedly, the "line too long" expressions for these aren't indenting quite correctly when the expression is in an ifdef block, note the indentation of `_tmp_152` here:

```
  always @(posedge clock) begin
      automatic logic _tmp_152;
    automatic logic [29:0] _T_145; 
    automatic logic        _T_146;     
    automatic logic [29:0] _T_147; 
    automatic logic        _T_148;     
    automatic logic        _T_149;     
    automatic logic        _T_150;     
    automatic logic        _T_151;      

    `ifndef SYNTHESIS 
      _T_145 = 30'h7FFF << {26'h0, |_T_36, |_T_38, |_T_40, _T_41}; 
      _T_146 = ~auto_in_ar_valid | _T_34 == ~(_T_145[14:0]) | reset;
      _T_147 = 30'h7FFF << {26'h0, |_T_74, |_T_76, |_T_78, _T_79};
      _T_148 = ~auto_in_aw_valid | _T_72 == ~(_T_147[14:0]) | reset;
      _tmp_152 = ~auto_in_aw_valid | auto_in_aw_bits_len == 8'h0;
      _T_149 = _tmp_152 | auto_in_aw_bits_size == 3'h3 | reset;
```

**Comment by lattner** [comment]
1bdba0b8 is technically enough to resolve this, but I'm going to grind on this a bit more.

**Comment by lattner** [comment]
Fixed in a much more general way in c25ae23e
